core/src/main/java/org/apache/brooklyn/core/effector/http/HttpCommandEffector.java

```diff
@@ -181,7 +204,19 @@ private HttpRequest buildHttpRequest(String httpVerb, URI uri, Map<String, Strin
                 if (contentType == null || contentType.equalsIgnoreCase(APPLICATION_JSON)) {
                     LOG.warn("Content-Type not specified. Using {}, as default (continuing)", APPLICATION_JSON);
                     body = toJsonString(payload);
-                } else if (!contentType.equalsIgnoreCase(APPLICATION_JSON)) {
+                } else if (contentType.equalsIgnoreCase(APPLICATION_X_WWW_FORM_URLENCODE)) {
+                    if (payload instanceof Map) {
+                        for (Map.Entry<String, String> entry : ((Map<String, String>) payload).entrySet()) {
+                            try {
+                                if (!body.equals("")) body += "&";
+                                body += URLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8.toString()) + "=" + URLEncoder.encode(entry.getValue(), StandardCharsets.UTF_8.toString());
```

Comment:
`URLEncoder`, despite its name, is for HTML form encoding, so it's probably not exactly what you want here, e.g. a space character will become a `+`.   I would probably use `java.net.URI`, e.g.
```
new URI(null,  "<img>",  null).toASCIIString()
```
returns `%3Cimg%3E`.